### DISCUSSION TITLE
White papers about using CFEngine and Ansible together

### DISCUSSION THREAD

**Discussion by vpodzime** [cfengine/core#4488] [Show and tell]
In the past few months, I'd been working on two white papers about using CFEngine and Ansible together, since this is quite common among our users. I'd like to share them here, in case some of you missed them.

The first one describes the different approaches and their unique strengths:
https://cfengine.com/company/blog-detail/ansiblecfengine/

The second white paper focuses on the scalability of the two tools, and shows some benchmarks at different scales:
https://cfengine.com/company/blog-detail/ansible-and-cfengine-scalability-white-paper/

**Comment by atsaloli** [comment]
I see https://cfengine.com/blog/2021/comparing-ansible-and-cfengine/ has been added since your original post.

All very nicely written.

I'm at an Ansible shop now... it's funny having to wait multiple minutes for a single Ansible run to complete (on a single host) when I am used to CFEngine agent runs taking less than a minute.

**Reply by nickanderson** [reply]
So, how would you like to use them together? Running Ansible modules from CFEngine? Or, is continual maintenance to avoid drift not something desired in that organization?

**Reply by atsaloli** [reply]
Yeah, continuous maintenance is not a thing (yet). Just initial build/configure.
